addons/account/report/account_invoice_report.py

```diff
@@ -56,131 +56,137 @@ def _compute_amounts_in_user_currency(self):
-                    count(*) * invoice_type.sign AS residual,
-                    ai.commercial_partner_id as commercial_partner_id,
-                    coalesce(partner.country_id, partner_ai.country_id) AS country_id
-        """
-        return select_str
-
-    def _from(self):
-        from_str = """
-                FROM account_invoice_line ail
-                JOIN account_invoice ai ON ai.id = ail.invoice_id
-                JOIN res_partner partner ON ai.commercial_partner_id = partner.id
-                JOIN res_partner partner_ai ON ai.partner_id = partner_ai.id
-                LEFT JOIN product_product pr ON pr.id = ail.product_id
-                left JOIN product_template pt ON pt.id = pr.product_tmpl_id
-                LEFT JOIN uom_uom u ON u.id = ail.uom_id
-                LEFT JOIN uom_uom u2 ON u2.id = pt.uom_id
-                JOIN (
-                    -- Temporary table to decide if the qty should be added or retrieved (Invoice vs Credit Note)
-                    SELECT id,(CASE
-                         WHEN ai.type::text = ANY (ARRAY['in_refund'::character varying::text, 'in_invoice'::character varying::text])
-                            THEN -1
-                            ELSE 1
-                        END) AS sign,(CASE
-                         WHEN ai.type::text = ANY (ARRAY['out_refund'::character varying::text, 'in_invoice'::character varying::text])
-                            THEN -1
-                            ELSE 1
-                        END) AS sign_qty
-                    FROM account_invoice ai
-                ) AS invoice_type ON invoice_type.id = ai.id
-        """
-        return from_str
-
-    def _group_by(self):
-        group_by_str = """
-                GROUP BY ail.id, ail.product_id, ail.account_analytic_id, ai.date_invoice, ai.id,
-                    ai.partner_id, ai.payment_term_id, u2.name, u2.id, ai.currency_id, ai.journal_id,
-                    ai.fiscal_position_id, ai.user_id, ai.company_id, ai.id, ai.type, invoice_type.sign, ai.state, pt.categ_id,
-                    ai.date_due, ai.account_id, ail.account_id, ai.partner_bank_id, ai.residual_company_signed,
-                    ai.amount_total_company_signed, ai.commercial_partner_id, coalesce(partner.country_id, partner_ai.country_id)
-        """
-        return group_by_str
+    @api.model
+    def _get_query_currency_table(self):
+        currency_rates = {c.id: c.rate for c in self.env.user.company_ids.mapped('currency_id')}
+        user_currency = self.env.company.currency_id
+        currency_table_args = [(c.id, currency_rates[user_currency.id] / currency_rates[c.currency_id.id], user_currency.decimal_places)
+                               for c in self.env.user.company_ids]
+        currency_table = ','.join('(%s, %s, %s)' % args for args in currency_table_args)
+        return '(VALUES %s) AS currency_table(company_id, rate, precision)' % currency_table
 
     @api.model_cr
     def init(self):
```

Comment:
there was a reason why the view initialization was made into different subqueries. Are you sure we don't need it anymore?